src/core/container.jl

```diff
@@ -185,12 +185,19 @@ function Libtask.consume(pc :: ParticleContainer)
     res
 end
 
-function weights(pc :: ParticleContainer)
+function weights(pc::ParticleContainer)
     @assert pc.num_particles == length(pc)
+
+    # compute the normalized weights
     logWs = pc.logWs
-    Ws = exp.(logWs .- maximum(logWs))
-    logZ = log(sum(Ws)) + maximum(logWs)
-    Ws = Ws ./ sum(Ws)
+    maximum_logWs = maximum(logWs)
+    Ws = @. exp(logWs - maximum_logWs)
+    sum_Ws = sum(Ws)
+    ldiv!(sum_Ws, Ws)
```

Comment:
Would you mind writing this as `Ws ./= sum_Ws`? Just a matter of taste, but it took me a minute to figure out what `ldiv!(sum_Ws, Ws)` is meant to do.

Reply:
I just assumed that `ldiv!` might be more optimized for such a simple operation than the broadcast machinery.  However, a quick benchmark shows that this seems to be the case only for matrices, for vectors both versions seem to be equally fast:
```julia
using BenchmarkTools
using LinearAlgebra

f!(A, b) = (A ./= b)

A = rand(10_000)
b = rand()

@btime f!($A, $b) # 5.731 μs (0 allocations: 0 bytes)
@btime ldiv!($b, $A) # 5.731 μs (0 allocations: 0 bytes)

A = rand(100, 100)

@btime f!($A, $b) # 11.469 μs (0 allocations: 0 bytes)
@btime ldiv!($b, $A) # 5.731 μs (0 allocations: 0 bytes)
```

Reply:
Huh, that's really weird... clearly your choice is better though, so I'm happy to go with it :)

Reply:
Same comment as Will, please change it to broadcast for readability. Happy to approve otherwise.